Updating the Transcript Rules

HelenBurge opened the pull request:
Are being updated from Task force feedback

Amending the rule from TF feedback: https://docs.google.com/spreadsheets/d/1OSkPFocXk4K3zYLnwS78WLsWO4PvE5yRcsauyefuIUI/edit#gid=1605073031

Closes issue(s):

Need for Call for Review:
This will require a 2 weeks Call for Review 

---

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Call for Review period. In case of disagreement, the longer period wins.

HelenBurge committed:
TranscriptRules

Are being updated from Task force feedback

HelenBurge committed:
Update audio-transcript-2eb176.md

HelenBurge committed:
Update audio-transcript-2eb176.md

HelenBurge committed:
Update video-transcript-1a02b0.md

Adding applicable updates from the TF feedback on audio transcripts

HelenBurge committed:
Update video-only-transcript-ee13b5.md

HelenBurge committed:
Update audio-transcript-2eb176.md

HelenBurge committed:
Update video-only-transcript-ee13b5.md

HelenBurge committed:
Update video-transcript-1a02b0.md

HelenBurge committed:
Merge branch 'develop' into TranscriptRules

Jym77 commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -5,6 +5,12 @@ rule_type: atomic
 accessibility_requirements:
+  wcag20:1.2.1: # Audio-only and Video-only (Prerecorded) (A)
+    forConformance: true
+    secondary: true
```
Should we start putting a reason here, as done in #2060 ?

HelenBurge replied:
I will assign these questions to @WilcoFiers as not sure on the secondary requirements needs.

WilcoFiers replied:
I'll handle it in my PR. Don't worry about it here. @HelenBurge maybe just remove this change from the PR and let me handle it on my PR.

Jym77 commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -5,6 +5,12 @@ rule_type: atomic
+  wcag20:1.2.1: # Audio-only and Video-only (Prerecorded) (A)
+    forConformance: true
+    secondary: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
```
Iirc, we do not have mappings for secondary requirements 🤔

HelenBurge replied:
I will assign these questions to @WilcoFiers as not sure on the secondary requirements needs.

WilcoFiers replied:
As above

Jym77 commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -24,12 +31,12 @@ acknowledgments:
 
-- playing; or,
-- has a [play button][] that is [visible][] and [included in the accessibility tree][].
+- autoplaying; or,
```
What does "autoplaying" means? [Audio or video element avoids automatically playing audio](https://www.w3.org/WAI/standards-guidelines/act/rules/80f0bf/proposed/) has a bunch of conditions for it.

Also, if the video plays but not automatically (e.g. after user interaction, but maybe that interaction hides the play button thus removing the second condition), would the rule become inapplicable?

HelenBurge replied:
I updated the PR description with how we decided on this in a meeting.

Jym77 replied:
I have no problem with using autoplay in the rule, but it needs to be somewhat defined or clarified. In #2030 and the following discussion, we've seen that "autoplay" is not as easy as we could think.

Does that refers to the `autoplay` attribute or something else? How do we handle the fact that browsers do not always respect it (due to engagement index, …)?

I think that "autoplaying" in somewhat ambiguous and needs to be clarified.

Jym77 commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -24,12 +31,12 @@ acknowledgments:
 ## Expectation
 
-The auditory information of each test target is available through a text transcript. That text transcript is [visible][] and [included in the accessibility tree][], either on the page or through a link.
+The auditory information of each test target is available through a text transcript. That text transcript is [visible][], either on the page or through a clearly labeled location.
```
```suggestion
The auditory information of each test target is available through a text transcript. That text transcript is [visible][], either on the page or through a [clearly labeled location][].
```
Plus add
```
[clearly labeled location]: #clearly-labeled-location 'Definition of Clearly Labeled Location'
```
in the references list at the end.

Jym77 commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -24,12 +31,12 @@ acknowledgments:
 ## Expectation
 
-The auditory information of each test target is available through a text transcript. That text transcript is [visible][] and [included in the accessibility tree][], either on the page or through a link.
+The auditory information of each test target is available through a text transcript. That text transcript is [visible][], either on the page or through a clearly labeled location.
```
I'm not sure why it is OK to have a transcript that is not included in the accessibility tree.
Wouldn't that be problematic, e.g., for Braille display users?

HelenBurge replied:
I believe it was due to the fact the transcript might not be included in the current page and therefore not in the accessibility tree.

Jym77 commented on _rules/video-only-transcript-ee13b5.md:
```diff
@@ -10,6 +10,12 @@ accessibility_requirements:
     inapplicable: further testing needed
+  wcag20:1.2.1: # Audio-only and Video-only (Prerecorded) (A)
+    forConformance: true
+    secondary: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
```
Same remark.

Jym77 commented on _rules/video-only-transcript-ee13b5.md:
```diff
@@ -36,7 +43,7 @@ This rule applies to any [non-streaming](#non-streaming-media-element) `video` e
 ## Expectation
 
-The visual information of each test target is available through a text transcript that is [visible][], [included in the accessibility tree][], and is either on the page or linked.
+The visual information of each test target is available through a text transcript that is [visible][], and is either on the page or in a clearly labeled location.
```
Same remarks.

For a transcript describing the visual content, the exclusion from the accessibility tree would also be problematic for screen reader users that have then no way to know what is the visual information in the video.

HelenBurge replied:
I believe it was due to the fact the transcript might not be included in the current page and therefore not in the accessibility tree.

Jym77 commented on _rules/video-transcript-1a02b0.md:
```diff
@@ -41,7 +42,7 @@ This rule applies to every [non-streaming](#non-streaming-media-element) `video`
 ## Expectation
 
-The visual information of each test target is available through a text transcript that is [visible][], [included in the accessibility tree][], and is either on the page or linked.
+The visual information of each test target is available through a text transcript that is [visible][], and is either on the page or in a clearly labeled location.
```
Same remarks.

Here also, non-accessible transcript is problematic for screen reader users.

HelenBurge replied:
I believe it was due to the fact the transcript might not be included in the current page and therefore not in the accessibility tree.

HelenBurge committed:
Update _rules/audio-transcript-2eb176.md

Co-authored-by: Jean-Yves Moyen <jym@siteimprove.com>

Jym77 commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -24,12 +31,12 @@ acknowledgments:
-- playing; or,
-- has a [play button][] that is [visible][] and [included in the accessibility tree][].
+- autoplaying; or,
+- has a [play button][] that is [visible][].
```
I do not really agree with the removal of the inclusion in the accessibility tree.
Looking at the survey report, the argument was that this might be a 1.3.1 failure rather than a 1.2.1. However, looking at the [Understanding document](https://www.w3.org/WAI/WCAG21/Understanding/audio-only-and-video-only-prerecorded.html) (which is not normative, but still very helpful) (emphasizes mine):
* The first sentence of the Intent is:
  > The intent of this Success Criterion is to make information conveyed by prerecorded audio-only and prerecorded video-only content **available to all users**.

  "available to all users" does imply that it is in the accessibility tree, since it would otherwise not be available to AT users.
  Especially, if a silent video has a hidden transcript, this does not provide any kind of alternative for blind users and I think that would fail 1.2.1.
* The first Benefit is:
  > This Success Criterion helps people who have **difficulty perceiving visual content**. Assistive technology can read text alternatives aloud, present them visually, or **convert them to braille**.

  People who have difficulties with visual content would not benefit at all from an alternative that is not included in the accessibility tree; as far as I know, braille displays also work with the accessibility tree and won't be able to present the alternative.
* The fourth benefit is:
  > People who are deaf-blind can read the text in braille.

  Again, not sure that this is possible if the alternative is not included in the accessibility tree.

I'm rather of the opinion that having an inaccessible alternative is actually a failure of 1.2.1 and thus should be part of this rule. This is maybe something that we need to discuss further.

daniel-montalvo commented:
Discussed at the TF meeting 29 Feb
https://www.w3.org/2024/02/29-wcag-act-minutes.html#t03

HelenBurge committed:
Update audio-transcript-2eb176.md

Updating the rules after a taskforce discussion

HelenBurge committed:
Update video-only-transcript-ee13b5.md

Updates from the taskforce meeting

HelenBurge committed:
Update video-transcript-1a02b0.md

Updated to match the task force amendments

HelenBurge committed:
Update audio-transcript-2eb176.md

HelenBurge committed:
Update video-only-transcript-ee13b5.md

Jym77 committed:
Merge branch 'develop' into TranscriptRules

Jym77 commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -5,6 +5,12 @@ rule_type: atomic
 accessibility_requirements:
+  wcag20:1.2.1: # Audio-only and Video-only (Prerecorded) (A)
+    forConformance: true
+    secondary: true
```
```suggestion
    secondary: This success criterion is **less strict** than this rule. This is because the rule does not consider that the audio may be a media alternative for text. Some of the failed examples satisfy this success criterion.
```

Jym77 commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -24,25 +31,27 @@ acknowledgments:
 
-- playing; or,
-- has a [play button][] that is [visible][] and [included in the accessibility tree][].
+- [autoplaying](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video#autoplay); or,
```
We normally avoid quoting from non-W3C sites, especially vendor ones. If this is just about the `autoplay` attribute like the "3 seconds" rule:
```suggestion
- the element has an `autoplay` [attribute value][] of `true`; or
```

Jym77 commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -24,25 +31,27 @@ acknowledgments:
 
 This rule applies to every [non-streaming](#non-streaming-media-element) `audio` element that is:
```
```suggestion
This rule applies to every [non-streaming](#non-streaming-media-element) `audio` element for which at least one of the following is true:
```
(taking the opportunity to update to how we write these now)

Jym77 commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -24,25 +31,27 @@ acknowledgments:
-- playing; or,
-- has a [play button][] that is [visible][] and [included in the accessibility tree][].
+- [autoplaying](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video#autoplay); or,
+- has a [play button][] that is [visible][].
```
```suggestion
- the element has a [play button][] that is [visible][].
```

Jym77 commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -24,25 +31,27 @@ acknowledgments:
 
 ## Background
 
+It is known that the user can set the browser to prevent autoplay. This rule assumes the browser has not been set to do this.
```
```suggestion
Some major browsers do not automatically play the 'video' unless the 'video' is muted. For these, the rule is never applicable. But for browsers that autoplay not muted video, the rule is still applicable.
```
(copying from #1655 for consistency)

Jym77 commented on _rules/audio-transcript-2eb176.md:
````diff
@@ -156,21 +165,6 @@ This `audio` element has native player controls and a [non-visible][visible] int
 </html>
 ```
 
-#### Failed Example 6
````
I'd like to have a Passed example with a transcript that is only visible but not included in the accessibility tree.
This serves as documentation that the CG and TF have reached the conclusion that it is not necessarily a failure of 1.2.1.

If inaccessible transcript is always a failure of 1.2.1, I don't think we should remove it for the rule. We can add 1.3.1 as a secondary requirement if needed (to allow implementations that report it to have a consistent implementation), or we need a separate rule checking that transcripts are accessible (but I don't see plan for that other rule yet 🤔).

HelenBurge replied:
The assumption states the transcript is part of the accessibility tree so is this still true?

kengdoj replied:
Agree to keep this example but as passed

HelenBurge replied:
Moved up :)

Jym77 commented on _rules/video-only-transcript-ee13b5.md:
```diff
@@ -10,6 +10,12 @@ accessibility_requirements:
     inapplicable: further testing needed
+  wcag20:1.2.1: # Audio-only and Video-only (Prerecorded) (A)
+    forConformance: true
+    secondary: true
```
```suggestion
    secondary: This success criterion is **less strict** than this rule. This is because the rule does not consider that the video may be a media alternative for text. Some of the failed examples satisfy this success criterion.
```

Jym77 commented on _rules/video-only-transcript-ee13b5.md:
```diff
@@ -36,19 +43,19 @@ This rule applies to any [non-streaming](#non-streaming-media-element) `video` e
 ## Assumptions
 
-A mechanism is available to start the video and that the video element is not simply used to display the [poster](https://www.w3.org/TR/html5/semantics-embedded-content.html#element-attrdef-video-poster).
+A mechanism is available to start the video and that the video element is not simply used to display the [poster](https://www.w3.org/TR/html5/semantics-embedded-content.html#element-attrdef-video-poster). This rule also assumes that if a transcript is present it is [included in the accessibility tree][].
```
Putting it as an assumption for something that is so easily testable sounds weird 🤔

HelenBurge replied:
This was what we decided was the best way to fix the concern. Basically, we are not testing the fact it is in the accessibility tree for 1.2.1 but assume it is there.

Jym77 commented on _rules/video-transcript-1a02b0.md:
```diff
@@ -41,20 +42,22 @@ This rule applies to every [non-streaming](#non-streaming-media-element) `video`
 
 ## Background
 
+It is known that the user can set the browser to prevent autoplay. This rule assumes the browser has not been set to do this.
```
```suggestion
Some major browsers do not automatically play the 'video' unless the 'video' is muted. For these, the rule is never applicable. But for browsers that autoplay not muted video, the rule is still applicable.
```
(but this rule doesn't seem to be about autoplay, so this probably shouldn't even be here in the first place)

Jym77 reviewed (changes requested):
I'm still not super happy with the inclusion in the accessibility tree part 😅 

---
From [the TF discussion](https://www.w3.org/2024/02/29-wcag-act-minutes.html#t03), I think I understand that TF reached the consensus that an inaccessible transcript is **not always** a 1.2.1 failure.

If this is indeed the case, then yes, we need to remove it (and the corresponding Failed Example). But in this case I'd like us to have a **Passed** example with an accessible transcript that is not failing 1.2.1. This will serve as documentation of this decision and essentially say: "no, inclusion in the accessibility tree is not required, look at this". (and if people disagree with this decision in the future, we'll have the example to foster concrete discussions instead of doing archaeology in Github history.

If I'm reading that wrong and that the issue with inaccessible transcripts is that they fail **both 1.2.1 and 1.3.1**, then I think that a better solution would be to add 1.3.1 as a **related** secondary requirement; this is similar to the [secondary requirement to 1.1.1 for "link have accessible name" (due to `area` elements)](https://github.com/act-rules/act-rules.github.io/blob/develop/_rules/link-non-empty-accessible-name-c487ae.md?plain=1#L28-L29)

HelenBurge committed:
Update _rules/audio-transcript-2eb176.md

Co-authored-by: Jean-Yves Moyen <jym@siteimprove.com>

HelenBurge committed:
Update _rules/audio-transcript-2eb176.md

Co-authored-by: Jean-Yves Moyen <jym@siteimprove.com>

HelenBurge committed:
Update _rules/audio-transcript-2eb176.md

Co-authored-by: Jean-Yves Moyen <jym@siteimprove.com>

HelenBurge committed:
Update _rules/audio-transcript-2eb176.md

Co-authored-by: Jean-Yves Moyen <jym@siteimprove.com>

HelenBurge committed:
Update _rules/audio-transcript-2eb176.md

Co-authored-by: Jean-Yves Moyen <jym@siteimprove.com>

HelenBurge committed:
Update _rules/video-only-transcript-ee13b5.md

Co-authored-by: Jean-Yves Moyen <jym@siteimprove.com>

HelenBurge committed:
Update _rules/video-transcript-1a02b0.md

Co-authored-by: Jean-Yves Moyen <jym@siteimprove.com>

HelenBurge committed:
Update video-only-transcript-ee13b5.md

HelenBurge committed:
Update video-only-transcript-ee13b5.md

HelenBurge committed:
Update video-only-transcript-ee13b5.md

HelenBurge committed:
Update video-transcript-1a02b0.md

Jym77 commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -5,6 +5,12 @@ rule_type: atomic
 description: |
   This rule checks that `audio` elements have a transcript that includes all auditory information.
 accessibility_requirements:
+  wcag20:1.2.1: # Audio-only and Video-only (Prerecorded) (A)
+    forConformance: true
+    secondary: This success criterion is **less strict** than this rule. This is because the rule does not consider that the audio may be a media alternative for text. Some of the failed examples satisfy this success criterion.
```
```suggestion
  wcag20:1.2.1: # Audio-only and Video-only (Prerecorded) (A)
    secondary: This success criterion is **less strict** than this rule. This is because the rule does not consider that the audio may be a media alternative for text. Some of the failed examples satisfy this success criterion.
```

Jym77 committed:
Fix frontmatters

daniel-montalvo commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -22,27 +25,29 @@ acknowledgments:
 
 ## Background
 
+Some major browsers do not automatically play the 'video' unless the 'video' is muted. For these, the rule is never applicable. But for browsers that autoplay not muted video, the rule is still applicable.
```
```suggestion
Some major browsers do not automatically play the 'video' unless it is muted, which makes the rule inapplicable in these browsers. The rule stil applies to browsers that autoplay not muted 'video'.
```

daniel-montalvo reviewed (dismissed):
Just minor editorials on my part

HelenBurge committed:
Update _rules/audio-transcript-2eb176.md

Co-authored-by: Daniel Montalvo <49305434+daniel-montalvo@users.noreply.github.com>

HelenBurge committed:
Update audio-transcript-2eb176.md

daniel-montalvo commented on _rules/video-transcript-1a02b0.md:
```diff
@@ -41,20 +42,22 @@ This rule applies to every [non-streaming](#non-streaming-media-element) `video`
 
 ## Background
 
+Some major browsers do not automatically play the 'video' unless the 'video' is muted. For these, the rule is never applicable. But for browsers that autoplay not muted video, the rule is still applicable.
```
```suggestion
Some major browsers do not automatically play the 'video' unless it is muted, which makes the rule inapplicable in these browsers. The rule still applies to browsers that autoplay not muted 'video'.
```
Sorry that I skipped this file earlier.

daniel-montalvo reviewed (dismissed):
Just adding the suggestion to the other occurrence of this paragraph, that I had skipped earlier.

Jym77 commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -22,27 +25,29 @@ acknowledgments:
 ## Assumptions
 
-There are no assumptions.
+This rule assumes that if a transcript is present it is [included in the accessibility tree][].
```
It feels more like a [Background note](https://www.w3.org/TR/act-rules-format/#background) ("information about the background for the development of the rule") than an [Assumption](https://www.w3.org/TR/act-rules-format/#assumptions) 🤔 (sorry 😅)

How about moving it to Background and phrasing it like:
```suggestion
Transcripts need to be available for all users, not just sighted users in order to satisfy WCAG. This rule does not test this, and it needs to be tested separately.
```

(and do the same with the other rules)

Jym77 reviewed (commented):
Not daring to reject it again… So just leaving it as a comment 😐

HelenBurge committed:
Update _rules/video-transcript-1a02b0.md

Co-authored-by: Daniel Montalvo <49305434+daniel-montalvo@users.noreply.github.com>

HelenBurge committed:
Update _rules/audio-transcript-2eb176.md

Co-authored-by: Jean-Yves Moyen <jym@siteimprove.com>

HelenBurge committed:
Update audio-transcript-2eb176.md

HelenBurge committed:
Update video-only-transcript-ee13b5.md

HelenBurge committed:
Update video-transcript-1a02b0.md

dan-tripp-siteimprove commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -22,27 +25,29 @@ acknowledgments:
 ## Assumptions
 
-There are no assumptions.
+Transcripts need to be available for all users, not just sighted users in order to satisfy WCAG. This rule does not test this, and it needs to be tested separately.
```
```suggestion
In order to satisfy WCAG, transcripts need to be available for all users, not just sighted users. This rule does not test this, and it needs to be tested separately.
```
Rearranging this sentence.  My reason is that with the current arrangement, someone might think that it means this: `In order to satisfy WCAG, transcripts only need to be available to sighted users.  But WCAG doesn't go far enough - really, transcripts need to be available to all users.`   That's not true.  My suggestion clears it up.

dan-tripp-siteimprove commented on _rules/video-only-transcript-ee13b5.md:
```diff
@@ -36,19 +39,19 @@ This rule applies to any [non-streaming](#non-streaming-media-element) `video` e
 ## Assumptions
 
-A mechanism is available to start the video and that the video element is not simply used to display the [poster](https://www.w3.org/TR/html5/semantics-embedded-content.html#element-attrdef-video-poster).
+Transcripts need to be available for all users, not just sighted users in order to satisfy WCAG. This rule does not test this, and it needs to be tested separately.
```
```suggestion
In order to satisfy WCAG, transcripts need to be available for all users, not just sighted users. This rule does not test this, and it needs to be tested separately.
```
Same as my similar comment from a minute ago.

dan-tripp-siteimprove commented on _rules/video-transcript-1a02b0.md:
```diff
@@ -41,20 +42,22 @@ This rule applies to every [non-streaming](#non-streaming-media-element) `video`
 ## Assumptions
 
-This rule assumes that a mechanism is available to start the video and that the video element is not simply used to display the [poster](https://www.w3.org/TR/html5/semantics-embedded-content.html#element-attrdef-video-poster).
+Transcripts need to be available for all users, not just sighted users in order to satisfy WCAG. This rule does not test this, and it needs to be tested separately.
```
```suggestion
In order to satisfy WCAG, transcripts need to be available for all users, not just sighted users. This rule does not test this, and it needs to be tested separately.
```
Same as my similar comment from a minute ago.

dan-tripp-siteimprove commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -22,27 +25,29 @@ acknowledgments:
 
 ## Background
 
+Some major browsers do not automatically play the 'video' unless it is muted, which makes the rule inapplicable in these browsers. The rule still applies to browsers that autoplay not muted 'video'.
```
```suggestion
Some major browsers do not automatically play the 'video' unless it is muted, which makes the rule inapplicable in these browsers. The rule still applies to browsers that autoplay unmuted 'video'.
```

dan-tripp-siteimprove commented on _rules/video-only-transcript-ee13b5.md:
```diff
@@ -36,19 +39,19 @@ This rule applies to any [non-streaming](#non-streaming-media-element) `video` e
 ## Background
 
-A "text transcript" in the context of this rule is defined in WCAG 2 as an [alternative for time based media](https://www.w3.org/TR/WCAG22/#dfn-alternative-for-time-based-media).
+A "text transcript" in the context of this rule is defined in WCAG 2 as an [alternative for time based media](https://www.w3.org/TR/WCAG22/#dfn-alternative-for-time-based-media). 
```
```suggestion
A "text transcript" in the context of this rule is defined in WCAG 2 as an [alternative for time-based media](https://www.w3.org/TR/WCAG22/#dfn-alternative-for-time-based-media). 
```

dan-tripp-siteimprove commented on _rules/video-transcript-1a02b0.md:
```diff
@@ -41,20 +42,22 @@ This rule applies to every [non-streaming](#non-streaming-media-element) `video`
 
 ## Background
 
+Some major browsers do not automatically play the 'video' unless it is muted, which makes the rule inapplicable in these browsers. The rule still applies to browsers that autoplay not muted 'video'.
```
```suggestion
Some major browsers do not automatically play the 'video' unless it is muted, which makes the rule inapplicable in these browsers. The rule still applies to browsers that autoplay unmuted 'video'.
```

HelenBurge replied:
amending this as got the spell checker unhappy!

kengdoj commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -22,27 +25,29 @@ acknowledgments:
-- playing; or,
-- has a [play button][] that is [visible][] and [included in the accessibility tree][].
+- the element has an `autoplay` [attribute value][] of `true`; or
+- the element has a [play button][] that is [visible][].
```
why was "and included in the accessibility tree" removed here?

HelenBurge replied:
Because from the survey discussion we agreed that to test the rule 1.2.1 the AT part is for a different checkpoint so it was made as an assumption instead: https://docs.google.com/spreadsheets/d/1OSkPFocXk4K3zYLnwS78WLsWO4PvE5yRcsauyefuIUI/edit#gid=1605073031

HelenBurge committed:
Update _rules/audio-transcript-2eb176.md

Co-authored-by: Dan Tripp <113939352+dan-tripp-siteimprove@users.noreply.github.com>

HelenBurge committed:
Update _rules/video-only-transcript-ee13b5.md

Co-authored-by: Dan Tripp <113939352+dan-tripp-siteimprove@users.noreply.github.com>

HelenBurge committed:
Update _rules/video-transcript-1a02b0.md

Co-authored-by: Dan Tripp <113939352+dan-tripp-siteimprove@users.noreply.github.com>

HelenBurge committed:
Update _rules/video-only-transcript-ee13b5.md

Co-authored-by: Dan Tripp <113939352+dan-tripp-siteimprove@users.noreply.github.com>

HelenBurge committed:
Update _rules/audio-transcript-2eb176.md

Co-authored-by: Dan Tripp <113939352+dan-tripp-siteimprove@users.noreply.github.com>

HelenBurge committed:
Update _rules/video-transcript-1a02b0.md

Co-authored-by: Dan Tripp <113939352+dan-tripp-siteimprove@users.noreply.github.com>

daniel-montalvo committed:
Merge branch 'develop' into TranscriptRules

HelenBurge committed:
Update video-transcript-1a02b0.md

Amended the updated of unmuted as the term is not recognised so used "not mute" instead.

HelenBurge committed:
Update audio-transcript-2eb176.md

HelenBurge committed:
Update video-transcript-1a02b0.md

kengdoj commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -22,27 +25,29 @@ acknowledgments:
-- playing; or,
-- has a [play button][] that is [visible][] and [included in the accessibility tree][].
+- the element has an `autoplay` [attribute value][] of `true`; or
+- the element has a [play button][] that is [visible][].
```
```suggestion
- the element has a [play button][] that is [visible][] and [included in the accessibility tree][].
```

HelenBurge committed:
Update _rules/audio-transcript-2eb176.md

Co-authored-by: Kathy Eng <kengdoj@users.noreply.github.com>

HelenBurge committed:
Update audio-transcript-2eb176.md

kengdoj commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -5,6 +5,8 @@ rule_type: atomic
 description: |
   This rule checks that `audio` elements have a transcript that includes all auditory information.
 accessibility_requirements:
```
Add SC 1.3.1 as a secondary requirement because it only applies when the transcript is visible.

Reference [Rules Format 1.1 Example 10](https://w3c.github.io/wcag-act/act-rules-format.html#example-cfb70282)

HelenBurge replied:
We did not include 1.3.1 as made the transcript as an assumption - I am not sure this should be done.

kengdoj replied:
Since we added Passed Ex 4, we would want to add 1.3.1 as secondary so an implementation that fails 1.3.1 is still consistent with this rule.

HelenBurge committed:
Update video-only-transcript-ee13b5.md

HelenBurge committed:
Update video-transcript-1a02b0.md

HelenBurge committed:
Update video-only-transcript-ee13b5.md

HelenBurge committed:
Update audio-transcript-2eb176.md

HelenBurge committed:
Update video-transcript-1a02b0.md

kengdoj commented on _rules/video-transcript-1a02b0.md:
````diff
@@ -145,5 +148,4 @@ A video element that is not [visible][].
 </html>
 ```
 
-[included in the accessibility tree]: #included-in-the-accessibility-tree 'Definition of included in the accessibility tree'
````
I'm not seeing this added back in yet so have unresolved this.

kengdoj commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -22,27 +25,29 @@ acknowledgments:
 
 ## Background
 
+Some major browsers do not automatically play the 'video' unless it is muted, which makes the rule inapplicable in these browsers. The rule still applies to browsers that autoplay an audible 'video'.
```
I don't think this is needed. The applicability includes a play button, so the rule could still apply if the audio does not autoplay.

kengdoj commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -22,27 +25,29 @@ acknowledgments:
 ## Assumptions
 
-There are no assumptions.
+In order to satisfy WCAG, transcripts need to be available for all users, not just sighted users. This rule does not test this, and it needs to be tested separately.
```
```suggestion
If a transcript is present, it is assumed it is in the accessibility tree. In order to satisfy WCAG, transcripts need to be available for all users, not just sighted users. This rule does not test this, and it needs to be tested separately.
```

kengdoj commented on _rules/video-only-transcript-ee13b5.md:
```diff
@@ -36,19 +39,19 @@ This rule applies to any [non-streaming](#non-streaming-media-element) `video` e
 ## Assumptions
 
-A mechanism is available to start the video and that the video element is not simply used to display the [poster](https://www.w3.org/TR/html5/semantics-embedded-content.html#element-attrdef-video-poster).
+In order to satisfy WCAG, transcripts need to be available for all users, not just sighted users. This rule does not test this, and it needs to be tested separately.
```
```suggestion
If a transcript is present, it is assumed it is in the accessibility tree. In order to satisfy WCAG, transcripts need to be available for all users, not just sighted users. This rule does not test this, and it needs to be tested separately.
```

kengdoj commented:
I couldn't add this in my review comments - Like the change you made in 2eb176, please change this failed example 4 to passed example 3 in ee13b5. 

#### Failed Example 4
This `video` element, which has no audio, has a text transcript available on the same page, but the transcript is not [included in the accessibility tree][].
```html
<html lang="en">
<video controls>
  <source src="/test-assets/rabbit-video/silent.mp4" type="video/mp4"></source>
  <source src="/test-assets/rabbit-video/silent.webm" type="video/webm"></source>
</video>
<p aria-hidden="true">The above video shows a giant fat rabbit climbing out of a hole in the ground.
He stretches, yawns, and then starts walking.
Then he stops to scratch his bottom.</p>
</html>
```

kengdoj commented on _rules/video-transcript-1a02b0.md:
```diff
@@ -41,20 +42,22 @@ This rule applies to every [non-streaming](#non-streaming-media-element) `video`
 ## Assumptions
 
-This rule assumes that a mechanism is available to start the video and that the video element is not simply used to display the [poster](https://www.w3.org/TR/html5/semantics-embedded-content.html#element-attrdef-video-poster).
+In order to satisfy WCAG, transcripts need to be available for all users, not just sighted users. This rule does not test this, and it needs to be tested separately.
```
```suggestion
If a transcript is present, it is assumed it is in the accessibility tree. In order to satisfy WCAG, transcripts need to be available for all users, not just sighted users. This rule does not test this, and it needs to be tested separately.
```

kengdoj commented on _rules/video-transcript-1a02b0.md:
```diff
@@ -41,20 +42,22 @@ This rule applies to every [non-streaming](#non-streaming-media-element) `video`
 
 ## Background
 
+Some major browsers do not automatically play the 'video' unless it is muted, which makes the rule inapplicable in these browsers. The rule still applies to browsers that autoplay an audible 'video'.
```
Autoplay isn't in the applicability, so I don't think this is needed.

HelenBurge replied:
Amended it to add the definition of the text transcript - as I think this is applicable here.

HelenBurge committed:
Update _rules/video-transcript-1a02b0.md

Co-authored-by: Kathy Eng <kengdoj@users.noreply.github.com>

HelenBurge committed:
Update _rules/video-only-transcript-ee13b5.md

Co-authored-by: Kathy Eng <kengdoj@users.noreply.github.com>

HelenBurge committed:
Update _rules/audio-transcript-2eb176.md

Co-authored-by: Kathy Eng <kengdoj@users.noreply.github.com>

HelenBurge committed:
Update audio-transcript-2eb176.md

HelenBurge committed:
Update video-only-transcript-ee13b5.md

Moved a failed to a passed and added a secondary 1.3.1

HelenBurge committed:
Update video-transcript-1a02b0.md

carlosapaduarte commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -22,27 +27,29 @@ acknowledgments:
 
-- playing; or,
-- has a [play button][] that is [visible][] and [included in the accessibility tree][].
+- the element has an [autoplay][] [attribute value][] of `true`; or
```
Since autoplay here is referring to the HTML attribute I don't think it needs to link to the definition.

carlosapaduarte commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -81,7 +88,7 @@ This `audio` element has native player controls and an external transcript.
 #### Passed Example 3
 
-This `audio` element has an `autoplay` attribute and an external transcript.
+This `audio` element has an [autoplay][] attribute and an external transcript.
```
Since autoplay here is referring to the HTML attribute I don't think it needs to link to the definition.

carlosapaduarte commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -131,7 +154,7 @@ This `audio` element has native player controls and an incorrect external transc
 #### Failed Example 4
 
-This `audio` element has an `autoplay` attribute and an incorrect external transcript.
+This `audio` element has an [autoplay][] attribute and an incorrect external transcript.
```
Since autoplay here is referring to the HTML attribute I don't think it needs to link to the definition.

carlosapaduarte commented on _rules/audio-transcript-2eb176.md:
```diff
@@ -22,27 +27,29 @@ acknowledgments:
 
 ## Background
 
+Some major browsers do not automatically play the 'video' unless it is muted, which makes the rule inapplicable in these browsers. The rule still applies to browsers that [autoplay][] an audible 'video'.
```
This one should link to the autoplay definition, but since it is being used in the Background section, the definition doesn't need to be objective.

carlosapaduarte reviewed (changes requested):
Comments about the autoplay attribute

HelenBurge commented:
After discussion, this pull request is no longer required.
